internal/apm/agent_configuration/read.go

```diff
@@ -0,0 +1,94 @@
+package agent_configuration
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/elastic/terraform-provider-elasticstack/generated/kbapi"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func (r *resourceAgentConfiguration) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state AgentConfiguration
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	kibana, err := r.client.GetKibanaOapiClient()
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to get Kibana client", err.Error())
+		return
+	}
+
+	apiResp, err := kibana.API.GetAgentConfigurationsWithResponse(
+		ctx,
+		&kbapi.GetAgentConfigurationsParams{
+			ElasticApiVersion: elasticAPIVersion,
+		},
+	)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get APM agent configurations", err.Error())
+		return
+	}
+	defer apiResp.Body.Close()
```

Comment:
Looks like `GetAgentConfigurationsWithResponse` returns own type (not `http.Response`) so GitHub Actions comment is correct and my previous suggestion is wrong. TL;DR - we need to remove line 37. Sorry for the confusion.